extension/background/youtubei.js

```diff
@@ -54,3 +60,109 @@ function parseTimestamp(ts) {
     const hours = parseInt(parts[2]) || 0
     return secs + (60 * mins) + (60 * 60 * hours)
 }
+
+// time-comments related code below
+
+const INNERTUBE_API_KEY = "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8"
+const INNERTUBE_CLIENT_NAME = "WEB"
+
+async function fetchChaptersFromComments(videoResponse) {
+    const comments = await fetchComments(videoResponse)
+
+    // Currently using only the first minimally suitable comment.
+    // Maybe later implement more sophisticated comment selection.
+    for (let i = 0; i < comments.length; i++) {
+        const tsContexts = getTimestampContexts(comments[i].text)
+        if (tsContexts.length) {
+            return tsContexts
+        }
+    }    
+
+    return []
+}
+
+function getTimestampContexts(text) {
+    const TIMESTAMP_PATTERN = /^((?:\d?\d:)?(?:\d?\d:)\d\d)\s(.+)$/
+    const chapters = []
+    const lines = text.split("\r\n")
+
+    for (let i = 0; i < lines.length; i++) {
+        const tsMatch = lines[i].match(TIMESTAMP_PATTERN)
+        if (!tsMatch) {
+            return []
+        }
+
+        const timestamp = tsMatch[1]
+        const title = tsMatch[2]
+        const time = parseTimestamp(timestamp)
+
+        chapters.push({
+            title,
+            timestamp,
+            time,
+        })
+    }
+
+    return chapters
+}
+
+async function fetchComments(videoResponse) {
+    const token = commentsContinuationToken(videoResponse)
+    if (!token) {
+        return []
+    }
+    const commentsResponse = await fetchNext(token)
+
+    const items = commentsResponse.onResponseReceivedEndpoints[1].reloadContinuationItemsCommand.continuationItems
+    if (!items) {
+        return []
+    }
+
+    const comments = []
+
+    for (const item of items) {
+        if (item.commentThreadRenderer) {
+            const cr = item.commentThreadRenderer.comment.commentRenderer
+            if (!cr.pinnedCommentBadge) { // Prioritize pinned comments
```

Comment:
It's better to do it in another place because this method just about fetching comments and not about prioritizing them.
You could just add `isPinned` property to the comment object.

Reply:
Done.